### PR TITLE
Update `kb_document` with the possibility to connect input to output

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description:
Add documentation of individual computations connecting its input and output.

**Note**: This implementation assumes that we only have one individual of each input dataset. A general implementation requires support from the workflow runner. 

## Type of change:
- [ ] Bug fix.
- [x] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand, including clearly named variables?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: oteapi_dlite/strategies/generate.py [hunk 1]
```diff
@@ -22,6 +22,15 @@
     from typing import Any
 
 
+# Constants
+hasInput = "https://w3id.org/emmo#EMMO_36e69413_8c59_4799_946c_10b05d266e22"
+hasOutput = "https://w3id.org/emmo#EMMO_c4bace1d_4db0_4cd3_87e9_18122bae2840"
+
+
+class KBError(ValueError):
+    """Invalud data in knowledge base."""
```
Comment (francescalb) [inline oteapi_dlite/strategies/generate.py:31]:
```suggestion
    """Invalid data in knowledge base."""
```

File: oteapi_dlite/strategies/generate.py [hunk 2]
```diff
@@ -174,6 +186,32 @@ class DLiteStorageConfig(AttrDict):
...
+                "\n\n"
+                "Note: This configuration relies on several assumptions:\n"
+                "  - The `kb_document_computation` class exists in the "
+                "knowledge base and is related to its input and output "
+                "dataset classes via `emmo:hasInput` and `emmo:hasOutput` "
+                "restrictions, respectively.\n"
+                "  - There exists only one individual of each input dataset "
+                "class.\n"
+                "  - There exists at most one individual of each output "
+                "dataset class.\n"
```
Comment (francescalb) [inline oteapi_dlite/strategies/generate.py:211]:
```suggestion
                "Requires `kb_document_class`, and is used to "
                "document the computation (model) that the "
                "individual (of `kb_document_class`) to be documented "
                "is output of."
                "When `kb_document_computation` is given a new individual of "
                "the computation subclass is created. Input and "
                "output datasets are documented using the relation "
                " `emmo:hasInput` and `emmo:hasOutput`, "
                "respectively.  The individual of `kb_document_class` is "
                "one of the output individuals."
                "\n\n"
                "Note: This configuration relies on several assumptions:\n"
                "  - The `kb_document_computation` class exists in the "
                "knowledge base and is related to its input and output "
                "dataset classes via `emmo:hasInput` and `emmo:hasOutput` "
                "restrictions, respectively.\n"
                "  - There exists only one individual of each input dataset "
                "class.\n"
                "  - There exists at most one individual of each output "
                "dataset class.\n"
```

File: oteapi_dlite/strategies/generate.py [hunk 3]
```diff
@@ -154,6 +163,9 @@ class DLiteStorageConfig(AttrDict):
             ),
         ),
     ] = None
+    kb_document_base_iri: Annotated[
+        str, Field(description="Base IRI or prefix for created individuals.")
+    ] = ":"
```
Comment (francescalb) [inline oteapi_dlite/strategies/generate.py:168]: Not included in tests.

Comment (jesper-friis) [reply oteapi_dlite/strategies/generate.py:168]: This configuration is always a string (not optional). Its default value is tested.

**Review by francescalb — APPROVED** [review]

**Review by francescalb — APPROVED** [review]

**Pull request merged**